cmd/hostpaths/hostpaths.go

```diff
@@ -216,28 +216,14 @@ func Start(ctx context.Context, options *context2.VirtualClusterOptions, init bo
 	return nil
 }
 
-func getVclusterObject(ctx context.Context, localManager manager.Manager, vclusterName, vclusterNamespace string, object client.Object) error {
-	err := localManager.GetClient().Get(ctx, types.NamespacedName{
-		Name:      vclusterName,
-		Namespace: vclusterNamespace,
-	}, object)
-	if err != nil {
-		return err
-	}
-
-	return nil
-}
-
-func getSyncerPodSpec(ctx context.Context, localManager manager.Manager, vclusterName, vclusterNamespace string) (*corev1.PodSpec, error) {
+func getSyncerPodSpec(ctx context.Context, kubeClient kubernetes.Interface, vclusterName, vclusterNamespace string) (*corev1.PodSpec, error) {
 	// try looking for the stateful set first
-	vclusterSts := &appsv1.StatefulSet{}
 
-	err := getVclusterObject(ctx, localManager, vclusterName, vclusterNamespace, vclusterSts)
+	vclusterSts, err := kubeClient.AppsV1().StatefulSets(vclusterNamespace).Get(ctx, vclusterName, metav1.GetOptions{})
 	if err != nil {
```

Comment:
up to you to do it or not since it's not exactly from this pr, 
but it'd be simpler to follow something like this 
```go
if kerrors.IsNotFound(err){
    ...
    return &vclusterDeploy.Spec.Template.Spec, nil
}else if err != nil {
   return nil, err
}
return &vclusterSts.Spec.Template.Spec, nil
```

Reply:
Yes. Applied this change